lib/nuntium/client.ex

```diff
@@ -13,6 +13,9 @@ defmodule Nuntium.Client do
     %Client{base_url: url, oauth2_client: oauth2_client}
   end
 
+  defp status_or_reason(%OAuth2.Error{reason: reason}), do: reason
+  defp status_or_reason(%{status_code: status_code}), do: status_code
+
```

Comment:
this is the same as in verboice channel. It is also the same logic that's underneath `parse_response` function (both in nuntium and verboice channel). I think parse_response could call this function for returning the error tuple instead of repeating the pattern again)

Reply:
Yes, I took this code from the Verboice Channel.

I think `parse_response` doesn't use this because it also adds an `:ok`/`:error` "header" in the response tuple, and you need to tell them apart.

There probably is some way to improve the code, but it's not straightforward, I think.